src/prefect/agent/docker/agent.py

```diff
@@ -518,20 +518,27 @@ def populate_env_vars(
         else:
             api = config.cloud.api
 
-        env = {}
+        env = {}  # type: ignore
         # Populate environment variables, later sources overriding
-
-        # 1. Logging config (optional)
-        # We only set this on non-DockerRun runs for backwards compatibility.
-        # In the future we don't want the agent to set logging config on a flow
-        # run unless explicitly asked.  We do this early on so later config
-        # sources can override
-        if run_config is None:
-            env.update({"PREFECT__LOGGING__LEVEL": config.logging.level})
-
-        # 2. Values set on the agent via `--env`
+        # 1. Values set on the agent via `--env`
         env.update(self.env_vars)
 
+        # 2. Logging config - if logging level is passed as an environment
+        # variable on the run config, it should take precedence
+        existing_logging_level = env.get(
+            "PREFECT__LOGGING__LEVEL", config.logging.level
+        )
```

Comment:
I'm curious if there was a reason for looking in the env?  If you set this env var, it should populate `config.logging.level` so it seems to me that setting `existing_logging_level = config.logging.level` is sufficient here

Reply:
Also very minor preference for calling this `default_logging_level`

Reply:
There seem to be three sources of logging level configs already implemented in the logic here, which I do not want to break.

1. `config.logging.level`, which aren't stored in `env` yet
2. The agent environment variables which now exist in `env` see `env.update(self.env_vars)` https://github.com/PrefectHQ/prefect/pull/4383/files/caacdb3e8693d1d1bf4e95663220e99d240a1ff2#diff-091912bb80855c71834fd22611ffb4d40919a954017024b5aa8588dac57b6ab4R524
3. `PREFECT__LOGGING__LEVEL` passed via the run config

Am I missing something here?

Reply:
I agree `existing` is a confusing name

Reply:
I'm a bit confused about the precedence here.

From low to high this looks like:
1. `--env` arguments passed to agent: may or may not be set
2. `config.logging.level`: will always have a value so the passed `--env PREFECT__LOGGING__LEVEL` is now ignored
3. `run_config.env`: may or may not be set, overrides all - this makes sense

Seems like 1/2 should be swapped

Reply:
1/2 are swapped in the get call, trying to get the env argument first and falling back to `config.logging.level`
```
existing_logging_level = env.get(
            "PREFECT__LOGGING__LEVEL", config.logging.level
        )
```

Seems like I just made things more confusing when I tried to make them more explicit, I'll take another pass at this

Reply:
I see. I would load things in order of low -> high and just let them override.